server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -360,9 +366,14 @@ public boolean isFieldAppliedFromTemplate(String name) {
 
     public void markFieldAsCopyTo(String fieldName) {
         copyToFields.add(fieldName);
+        if (mappingLookup.isSourceSynthetic()) {
+            // Mark this field as containing copied data
+            // meaning it should not be present in synthetic _source (to be consistent with stored _source).
+            ignoredFieldValues.add(IgnoredSourceFieldMapper.NameValue.fromContext(this, fieldName, XContentDataHelper.nothing()));
```

Comment:
Nit: add a note here that ignored fields take precedence, overriding any value that would be injected from the corresponding fields?

Reply:
Note also that we want this to happen even if the current context has been recorded to `_ignored_source`, hence no check for `clonedSource`.